pkg/client/client.go

```diff
@@ -52,159 +38,90 @@ const (
-	config := sess.Config
-	tokenRefresher, err := authentication.NewIAMAuthRepository(config, &rest.Client{
-		DefaultHeader: gohttp.Header{
-			"User-Agent": []string{http.UserAgent()},
-		},
-	})
-	if err != nil {
-		return err
-	}
-	return tokenRefresher.AuthenticateAPIKey(config.BluemixAPIKey)
 }
 
 type User struct {
-	ID         string
-	Email      string
-	Account    string
-	cloudName  string `default:"bluemix"`
-	cloudType  string `default:"public"`
-	generation int    `default:"2"`
-}
-
-func fetchUserDetails(sess *bxsession.Session, generation int) (*User, error) {
-	config := sess.Config
-	user := User{}
-	var bluemixToken string
-
-	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
-		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
-	} else {
-		bluemixToken = config.IAMAccessToken
-	}
-
-	token, err := jwt.Parse(bluemixToken, func(token *jwt.Token) (interface{}, error) {
-		return "", nil
-	})
-	if err != nil && !strings.Contains(err.Error(), "key is of invalid type") {
-		return &user, err
-	}
-
-	claims := token.Claims.(jwt.MapClaims)
-	if email, ok := claims["email"]; ok {
-		user.Email = email.(string)
-	}
-	user.ID = claims["id"].(string)
-	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
-	iss := claims["iss"].(string)
-	if strings.Contains(iss, "https://iam.cloud.ibm.com") {
-		user.cloudName = "bluemix"
-	} else {
-		user.cloudName = "staging"
-	}
-	user.cloudType = "public"
-
-	user.generation = generation
-	return &user, nil
+	ID      string
+	Email   string
+	Account string
 }
 
-func NewClient(apikey, ep string, debug bool) (*Client, error) {
+func NewClient(apikey string, ep map[string]string, debug bool) (*Client, error) {
 	c := &Client{}
 
-	bxSess, err := bxsession.New(&bluemix.Config{
-		BluemixAPIKey:         apikey,
-		TokenProviderEndpoint: &ep,
-		Debug:                 debug,
-	})
+	auth, err := core.GetAuthenticatorFromEnvironment(serviceIBMCloud)
 	if err != nil {
 		return nil, err
 	}
 
-	c.Session = bxSess
-
-	err = authenticateAPIKey(bxSess)
+	iamv1, err := iamidentityv1.NewIamIdentityV1(&iamidentityv1.IamIdentityV1Options{
+		Authenticator: auth,
+		URL:           ep["TPEndpoint"],
+	})
 	if err != nil {
 		return nil, err
 	}
-
-	c.User, err = fetchUserDetails(bxSess, 2)
+	apiKeyDetailsOpt := iamidentityv1.GetAPIKeysDetailsOptions{IamAPIKey: ptr.To(apikey)}
+	apiKey, _, err := iamv1.GetAPIKeysDetails(&apiKeyDetailsOpt)
 	if err != nil {
 		return nil, err
 	}
-
-	ctrlv2, err := controllerv2.New(bxSess)
-	if err != nil {
-		return nil, err
+	if apiKey == nil {
+		return nil, fmt.Errorf("could not retrieve account id")
 	}
-
-	ctrlv1, err := controller.New(bxSess)
-	if err != nil {
-		return nil, err
+	c.User = &User{
+		ID:      *apiKey.ID,
+		Account: *apiKey.AccountID,
 	}
-
-	catalogClient, err := catalog.New(bxSess)
+	resourceControllerOptions := &resourcecontrollerv2.ResourceControllerV2Options{
+		URL:           ep["RCEndpoint"],
```

Comment:
I think its better to declare and use the constant.

Reply:
Addressed. Thanks!